src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2594,11 +2594,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
         Protected Overrides Function CommonCreateTupleTypeSymbol(elementTypes As ImmutableArray(Of ITypeSymbol), elementNames As ImmutableArray(Of String)) As INamedTypeSymbol
-            Throw New NotSupportedException(VBResources.TuplesNotSupported)
+            Dim vbElementTypes = elementTypes.SelectAsArray(Function(t) t.EnsureVbSymbolOrNothing(Of TypeSymbol)(NameOf(elementTypes)))
```

Comment:
I will add additional validation.
For now this change implements the case where names are all-or-none. I would be easier to relax that as a separate change.

---
In reply to: [71412641](https://github.com/dotnet/roslyn/pull/12596#discussion_r71412641) [](ancestors = 71412641)